### PR TITLE
Issue #877: fix TreeWalkerTest JUnitTestsShouldIncludeAssert violations

### PR DESCRIPTION
**Pull request by Nimfadora**
Issue #877

### CODE REVIEW

**Comment by Nimfadora** [comment]
@romani cobertura fails on this test

**Comment by romani** [comment]
@Nimfadora ,

please make such changes on your local:
```
 $ git diff
diff --git a/pom.xml b/pom.xml
index 354f846..7f06041 100644
--- a/pom.xml
+++ b/pom.xml
@@ -1502,7 +1502,7 @@
             <artifactId>cobertura-maven-plugin</artifactId>
             <version>2.7</version>
             <configuration>
-              <quiet>true</quiet>
+              <!--<quiet>true</quiet>-->
               <formats>
```

it might tell about some problems that are related your changes

**Comment by romani** [comment]
@Nimfadora , please rebase your PR over latest master, it will print you missed coverage in build log.

**Comment by romani** [comment]
please rebase on latest master. 
after each release it is mandatory to rebase to avoid CIs failures.

**Comment by Nimfadora** [comment]
@romani 
when I rebased and start cobertura with `quiet` turned off, it shows just nothing but:

```
[Thread-2] ERROR net.sourceforge.cobertura.coveragedata.TouchCollector - Cannot apply touches
java.lang.NoSuchMethodException: com.puppycrawl.tools.checkstyle.api.AutomaticBean.__cobertura_classmap(net.sourceforge.cobertura.coveragedata.LightClassmapListener)
        at java.lang.Class.getDeclaredMethod(Class.java:2130)
        at net.sourceforge.cobertura.coveragedata.TouchCollector.applyTouchesToSingleClassOnProjectData(TouchCollector.java:127)
        at net.sourceforge.cobertura.coveragedata.TouchCollector.applyTouchesOnProjectData(TouchCollector.java:110)
        at net.sourceforge.cobertura.coveragedata.ProjectData.saveGlobalProjectData(ProjectData.java:272)
        at net.sourceforge.cobertura.coveragedata.SaveTimer.run(SaveTimer.java:33)
        at java.lang.Thread.run(Thread.java:745)
[Thread-2] ERROR net.sourceforge.cobertura.coveragedata.TouchCollector - Cannot apply touches
java.lang.NoSuchMethodException: com.puppycrawl.tools.checkstyle.api.AbstractViolationReporter.__cobertura_classmap(net.sourceforge.cobertura.coveragedata.LightClassmapListener)
        at java.lang.Class.getDeclaredMethod(Class.java:2130)
        at net.sourceforge.cobertura.coveragedata.TouchCollector.applyTouchesToSingleClassOnProjectData(TouchCollector.java:127)
        at net.sourceforge.cobertura.coveragedata.TouchCollector.applyTouchesOnProjectData(TouchCollector.java:110)
        at net.sourceforge.cobertura.coveragedata.ProjectData.saveGlobalProjectData(ProjectData.java:272)
        at net.sourceforge.cobertura.coveragedata.SaveTimer.run(SaveTimer.java:33)
        at java.lang.Thread.run(Thread.java:745)
[Thread-2] ERROR net.sourceforge.cobertura.coveragedata.TouchCollector - Cannot apply touches
java.lang.NoSuchMethodException: com.puppycrawl.tools.checkstyle.api.AbstractCheck.__cobertura_classmap(net.sourceforge.cobertura.coveragedata.LightClassmapListener)
        at java.lang.Class.getDeclaredMethod(Class.java:2130)
        at net.sourceforge.cobertura.coveragedata.TouchCollector.applyTouchesToSingleClassOnProjectData(TouchCollector.java:127)
        at net.sourceforge.cobertura.coveragedata.TouchCollector.applyTouchesOnProjectData(TouchCollector.java:110)
        at net.sourceforge.cobertura.coveragedata.ProjectData.saveGlobalProjectData(ProjectData.java:272)
        at net.sourceforge.cobertura.coveragedata.SaveTimer.run(SaveTimer.java:33)
        at java.lang.Thread.run(Thread.java:745)

```
But the same errors I get on the master branch when doing clean verify or cobertura:check.

**Comment by romani** [comment]
@Nimfadora , it is #4450 . Do you still experiencing this problem on master and on all branches that you work on ?
If yes, we need to focus on #4450 , asap as it is blocker for you.

**Comment by Nimfadora** [comment]
@romani it appears from time to time. There are moments when all passes normally, the other day it fails.

**Comment by romani** [comment]
@Nimfadora , I do not see changes in pmd config.
Is it the last update for this issue? Or we have more problematic cases.

File: src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java [hunk 1]
```diff
@@ -183,11 +185,33 @@ public void testForInvalidCheckImplementation() throws Exception {
...
+        catch (CheckstyleException ex) {
+            assertEquals("Invalid exception message",
+                "MismatchedTokenException occurred during the analysis of file input.java.",
+                ex.getMessage());
+        }
+        try {
+            treeWalker.processFiltered(new File("input.txt"), lines);
+        }
+        catch (CheckstyleException ex) {
+            fail("Exception is not expected, file with txt extension should not be processed");
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java]: We said before catching an exception and just failing is bad practice.
Exception should be populated to junit runner to display to user.

Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java]: The situation is as follows: method processFiltered if called from BaseCheckTestSupport [this](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java#L170) condition is not checked as it checks it before [here](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheck.java#L78).
So I have to call it on TreeWalker class, and check that on the same lines but different extensions it fails on java file and does not on txt. What shoul I pass to junit runner and for what purpose, if the cause of test failure is not in the exception but in the fact that it is thrown?

**Comment by rnveach** [comment]
> What should I pass to junit runner and for what purpose, if the cause of test failure is not in the exception but in the fact that it is thrown?

Just allow the exception to be populated all the way up he chain. Removing the `try-catch-fail` is more than enough. The runner will see the exception and display it to the user. With your `try-catch-fail` user will get a bland message and not see the exception and stacktrace. All exceptions are unexpected so the runner will fail if any exception is caught by it.

**Pull request merged**